packages/netlify-cms-core/src/backend.ts

```diff
@@ -561,35 +599,93 @@ export class Backend {
     };
   }
 
-  unpublishedEntries(collections: Collections) {
-    return this.implementation.unpublishedEntries!()
-      .then(entries =>
-        entries.map(loadedEntry => {
-          const collectionName = loadedEntry.metaData!.collection;
+  async processUnpublishedEntry(
+    collection: Collection,
+    entryData: UnpublishedEntry,
+    withMediaFiles: boolean,
+  ) {
+    const { slug } = entryData;
+    let extension: string;
+    if (collection.get('type') === FILES) {
+      const file = collection.get('files')!.find(f => f?.get('name') === slug);
+      extension = extname(file.get('file'));
+    } else {
+      extension = selectFolderEntryExtension(collection);
+    }
+    const dataFiles = sortBy(
+      entryData.diffs.filter(d => d.path.endsWith(extension)),
+      f => f.path.length,
+    );
+    // if the unpublished entry has no diffs, return the original
+    let data = '';
+    let newFile = false;
+    let path = slug;
+    if (dataFiles.length <= 0) {
+      const loadedEntry = await this.implementation.getEntry(
+        selectEntryPath(collection, slug) as string,
+      );
+      data = loadedEntry.data;
+      path = loadedEntry.file.path;
+    } else {
+      const entryFile = dataFiles[0];
+      data = await this.implementation.unpublishedEntryDataFile(
+        collection.get('name'),
+        entryData.slug,
+        entryFile.path,
+        entryFile.id,
+      );
+      newFile = entryFile.newFile;
+      path = entryFile.path;
+    }
+
+    const mediaFiles: MediaFile[] = [];
+    if (withMediaFiles) {
```

Comment:
We only read media files when loading the entry in the editor.
In the the workflow view we only read the content file.